src/main.rs

```diff
@@ -7,40 +7,32 @@ extern crate pgs_files;
 extern crate log;
 extern crate kernlog;
 
-use std::thread::spawn;
 use std::env;
-use std::fs::{File, create_dir_all, metadata};
+use std::fs::{create_dir_all, metadata, File};
+use std::io::{self, Error, Write};
 use std::os::unix::fs::symlink;
-use std::io::Write;
 use std::path::Path;
+use std::process::exit;
+use std::thread::spawn;
 
 use cronparse::crontab::{AnacrontabEntry, SystemCrontabEntry, UserCrontabEntry};
 
 mod generate;
 mod process;
 
 include!(concat!(env!("OUT_DIR"), "/config.rs"));
-static SYSTEM_CRONTAB_DIR: &'static str = "/etc/cron.d";  // SystemCrontabEntry
-static SYSTEM_CRONTAB_FILE: &'static str = "/etc/crontab";
-static ANACRONTAB_FILE: &'static str = "/etc/anacrontab";  // AnacrontabEntry
-static REBOOT_FILE: &'static str = "/run/crond.reboot";
-
-macro_rules! try_ {
-    ($exp:expr) => {
-        match $exp {
-            Ok(value) => value,
-            Err(err) => { error!("{}", err); return; }
-        }
-    }
-}
+static SYSTEM_CRONTAB_DIR: &str = "/etc/cron.d"; // SystemCrontabEntry
+static SYSTEM_CRONTAB_FILE: &str = "/etc/crontab";
+static ANACRONTAB_FILE: &str = "/etc/anacrontab"; // AnacrontabEntry
+static REBOOT_FILE: &str = "/run/crond.reboot";
 
-fn main() {
+fn main() -> Result<(), Error> {
     log::set_logger(|filter| kernlog::KernelLog::init_level(log::LogLevelFilter::Error, filter)).unwrap();
 
     let dest_dir = match env::args().nth(1) {
         None => {
             println!("Usage: systemd-crontab-generator <destination-directory>");
-            return;
+            exit(1);
```

Comment:
I think it's more idiomatic to return some `Err(_)` here since we already have `Result<(), Error>` return type.

Reply:
Yes, but it raises further questions. For one, what Error type? This PR has the Err result being type io::Error. If we add `return Err(Error::new(ErrorKind::InvalidInput, "Invalid Input"));` then we get:
```
Usage: systemd-crontab-generator <destination-directory>
Error: Custom { kind: InvalidInput, error: "Invalid Input" }
```
which seems worse to me, or at least different, where my intent was to not change behavior.

See https://doc.rust-lang.org/edition-guide/rust-2018/error-handling-and-panics/question-mark-in-main-and-tests.html#more-details for more details.

So I take your point, but I don't think main returning a Result means that `exit(1)` is no longer a reasonable thing to do.

If you wanted to get fancy the use of clap or structopt crates might be something to look into.